### PR TITLE
add auth design doc

### PR DESCRIPTION
**Pull request by nlandolfi**
See doc, let me know thoughts comments

### CODE REVIEW

File: auth/design.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Authorizing is the process of subsequently including a session token along with actions you take against the elos system. You have already been authenticated (you have a session token), but you may still not be allowed to take the action. Therefore, you must, of course, include the tokens.
+
+ Keep in mind:
+  1. NEVER include the session token in a URL
+  2. NEVER log the session token
+  3. ALLOWED to store the session token in the client (unencrypted).
+  4. ALWAYS (for HTTP) include the session token as a header.
+
+Authorization is required for _any_ communication to the elos server. It is _not_ something that you only add to particulary payloads. Declaring identity is DEFAULT and required. Therefore the session token should be present in the header, as a delcariation of identity and a certificate of authentication.
```
Comment (viggyfresh) [inline auth/design.md]: typo in declaration

File: auth/design.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### Relation to Auth Service
+
+The elos Auth service housed in this (auth) directory is the source of truth for authenticating and thereafter, authorizing a user. The AuthenticateRequest handles both the case of public, private credential and also the case of session authorization (public=session_token, private=<empty>).
+
+To any elos service, at the outer layer of the elos graph, authorization and authentication are the same thing: "who is this person talking to me?" The response to a failure of authentication/authorization is also the same: "don't talk to this client."
+
+Note, however, that authentication as described above is purely to CREATE a session. Sessions are safe forms of authorization for third-party clients, since they expire and are more disposable than credentials. Technically, one could authorize with a (public,private) credential pair every time, but this is foolish and impractical. Because a client must NOT store a users's credentials, it is impossible to implement.
```
Comment (viggyfresh) [inline auth/design.md]: space after public,

File: auth/design.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+### Authorizing
+
+Authorizing is the process of subsequently including a session token along with actions you take against the elos system. You have already been authenticated (you have a session token), but you may still not be allowed to take the action. Therefore, you must, of course, include the tokens.
+
+ Keep in mind:
+  1. NEVER include the session token in a URL
+  2. NEVER log the session token
+  3. ALLOWED to store the session token in the client (unencrypted).
+  4. ALWAYS (for HTTP) include the session token as a header.
```
Comment (viggyfresh) [inline auth/design.md]: what form should that header take?

Comment (nlandolfi) [inline auth/design.md]: Elos-Session-Token ??

Comment (viggyfresh) [inline auth/design.md]: 👍

File: auth/design.md [hunk 4]
```diff
@@ -26,12 +26,12 @@ Authorizing is the process of subsequently including a session token along with
   3. ALLOWED to store the session token in the client (unencrypted).
   4. ALWAYS (for HTTP) include the session token as a header.
 
-Authorization is required for _any_ communication to the elos server. It is _not_ something that you only add to particulary payloads. Declaring identity is DEFAULT and required. Therefore the session token should be present in the header, as a delcariation of identity and a certificate of authentication.
+Authorization is required for _any_ communication to the elos server. It is _not_ something that you only add to particulary payloads. Declaring identity is DEFAULT and required. Therefore the session token should be present in the header, as a delcaration of identity and a certificate of authentication.
```
Comment (viggyfresh) [inline auth/design.md]: *declaration 😄

Comment (nlandolfi) [inline auth/design.md]: ahhhhhhh

Comment (nlandolfi) [inline auth/design.md]: lol my bad

Comment (nlandolfi) [inline auth/design.md]: done

**Comment by nlandolfi** [comment]
btw @viggyfresh i think you have to positively approve this for it to be green

**Pull request merged**